hibernate-core/src/test/java/org/hibernate/test/cid/CompositeIdTest.java

```diff
@@ -108,11 +105,11 @@ public void testDistinctCountOfEntityWithCompositeId() {
 		try {
 			long count = ( Long ) s.createQuery( "select count(distinct o) FROM Order o" ).uniqueResult();
 			if ( ! getDialect().supportsTupleDistinctCounts() ) {
-				fail( "expected SQLGrammarException" );
+				fail( "expected PersistenceException" );
 			}
 			assertEquals( 2l, count );
 		}
-		catch ( SQLGrammarException e ) {
+		catch ( PersistenceException e ) {
```

Comment:
@simkam 

I think this issue is due to the new exception handling but I think we should expect a `SQLGrammarException` wrapped in a `PersistenceException`.

Could you try with this change instead:
```
catch ( PersistenceException e ) {
    if ( !(e.getCause() instanceof SQLGrammarException) || getDialect().supportsTupleDistinctCounts() ) {
        throw e;
    }
}
```

Reply:
yep, it works as well, PR updated